client_test/rpc_test.go

```diff
@@ -83,18 +79,56 @@ func TestRpcClient(t *testing.T) {
 	assert.Regexp(t, "VirtualFund.0x.*", vRes.Id)
 
 	rpcClientA.WaitForObjectiveCompletion(vRes.Id)
-	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, vRes.Id)
-	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, vRes.Id)
+	rpcClientB.WaitForObjectiveCompletion(vRes.Id)
+	rpcClientI.WaitForObjectiveCompletion(vRes.Id)
+
 	rpcClientA.Pay(vRes.ChannelId, 1)
 
 	closeVId := rpcClientA.CloseVirtual(vRes.ChannelId)
 	rpcClientA.WaitForObjectiveCompletion(closeVId)
-	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, closeVId)
-	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, closeVId)
+	rpcClientB.WaitForObjectiveCompletion(closeVId)
+	rpcClientI.WaitForObjectiveCompletion(closeVId)
 
 	closeId := rpcClientA.CloseLedger(res.ChannelId)
-
 	rpcClientA.WaitForObjectiveCompletion(closeId)
-	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, closeId)
+	rpcClientI.WaitForObjectiveCompletion(closeId)
 
 }
+
+// setupNitroNodeWithRPCClient is a helper function that spins up a Nitro Node RPC Server and returns an RPC client connected to it.
+func setupNitroNodeWithRPCClient(
+	pk []byte,
+	msgPort int,
+	rpcPort int,
+	chain *chainservice.MockChainService,
+	logDestination *os.File,
+) (*rpc.RpcClient, *p2pms.P2PMessageService, func()) {
+	chainId, err := chain.GetChainId()
+	if err != nil {
+		panic(err)
```

Comment:
Not a big deal here, but, in general, I would prefer to bubble up errors to the outermost caller when possible.

Reply:
I'm with you there, but was planning on hiding behind the "it's only test code" defence :-P.

Reply:
It seems like the preferred pattern is to use `t.Fatalf` to fail a test. Bubbling up an error (even in test code) allows us to follow that pattern.